components/Header.js

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import Link from "next/link";
+import Router from "next/router";
+
+import MenuToggle from "./MenuToggle";
+import Menu from "./Menu";
+
+class Header extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { menuVisible: false };
+    this.toggleMenu = this.toggleMenu.bind(this);
+    this.closeMenu = this.closeMenu.bind(this);
+    Router.events.on("routeChangeComplete", this.closeMenu);
```

Comment:
Po przeładowaniu strony i użyciu menu po raz pierwszy dostaję w konsoli przeglądarki warning: 

> "Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method."

Korzystając z https://github.com/zeit/next.js/#router-events i treści warninga da się to naprawić dodając metodę

`  
componentWillUnmount(){
    Router.events.off('routeChangeComplete', this.closeMenu)
}
`